sforzando/q-lako
# Implement duplicate judgment

## Describe the PR

To close #35 .
I need the ability to notify you of the presence of duplicate items.

## Screenshots

<img width="1860" alt="Screen Shot 2020-12-11 at 21 06 09" src="https://user-images.githubusercontent.com/67737076/101901638-b33b5080-3bf4-11eb-8978-78138182454f.png">

## Describe the solution you'd like

### main.py

1. Get the `keyword` and `product_group`of the selected item
1. get a list of items registered in the Airtable that match `product_group`
1. Compare the keywords with the titles of the lists retrieved from Airtable and retrieve the similar ones by ambiguous search
1. Convert the retrieved list into an Asset class
1. if the search is successful, similar items will be displayed at the specified location and a `flash message` will be displayed

### airtable_client.py

```airtable_client.py

    def get_similar_items_by_keyword(self, product_group: str, keyword: str):
        """Fetch Airtable item list.

        It retrieves items with a common product group stored in the Airtable and
        outputs similar items related to the keyword.

        Args:
            keyword (str): Keywords used in the search.
            product_group (str): Product group name for the specified item.

        Returns:
            similar_items (list<Asset>): List of Asset classes for similar items.
        """

```

### registration.html

```registration.html

{% if similar_items %}
<div>
  <p>Similar items:</p>
  <div">
    {% for item in similar_items %}
    <div">
      <p>
        <span>Title: </span>{{ item["title"] }}
      </p>
      <p">
        <span>Registered by: </span>{{ item["registrant_name"] }}
        <span>at: </span>{{ item["registered_at"]}}
      </p>
    </div>
    {% endfor %}
  </div>
</div>
{% endif %}

```

## To reproduce

Steps to check the behavior:

1. Prepare `Python` environment
    `source venv/bin/activate`
    `pip install --upgrade -r requirements.txt`
1. Run `python main.py`
1. Access to `http://localhost:8888/`
1. Enter a `Python`
1. Select the item containing the `Python` string
1. Make sure you see similar items under the `View on Amazon` button

## Additional context

None.
